### PR TITLE
DiagnosticSourceListener generates it's own nuget package

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
#116 and #79 
Also #62 and #82

### CODE REVIEW

File: src/Adapters/DiagnosticSource/DiagnosticSourceListener.csproj [hunk 1]
```diff
@@ -0,0 +1,76 @@
+﻿
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Company>Microsoft</Company>
+    <Copyright>Copyright © Microsoft. All Rights Reserved.</Copyright>
```
Comment (Dmitry-Matveev) [inline src/Adapters/DiagnosticSource/DiagnosticSourceListener.csproj]: Minor: I remember guidelines saying that copyright should be "© Microsoft Corporation. All rights reserved."

File: src/Adapters.Tests/DiagnosticSource.Tests/DiagnosticSourceListener.Tests.csproj [hunk 2]
```diff
@@ -18,19 +18,17 @@
...
   <ItemGroup>
     <!-- Identifies the project as test project -->
     <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
   </ItemGroup>
 
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), 'Test.props'))\Test.props" />
 
-  <ItemGroup>
-    <ProjectReference Include="..\..\Adapters\DiagnosticSource\DiagnosticSource.csproj">
-      <IncludeAssets>All</IncludeAssets>
```
Comment (Dmitry-Matveev) [inline src/Adapters.Tests/DiagnosticSource.Tests/DiagnosticSourceListener.Tests.csproj]: Do we need this in new reference as well?

File: src/Adapters/Log4NetAppender.Net45/ApplicationInsightsAppender.cs [hunk 3]
```diff
@@ -142,7 +142,7 @@ private void SendTrace(LoggingEvent loggingEvent)
         private void BuildCustomProperties(LoggingEvent loggingEvent, ITelemetry trace)
         {
             trace.Timestamp = loggingEvent.TimeStamp;
-            trace.Context.User.Id = loggingEvent.UserName;
+            trace.Context.User.AuthenticatedUserId = loggingEvent.UserName;
```
Comment (Dmitry-Matveev) [inline src/Adapters/Log4NetAppender.Net45/ApplicationInsightsAppender.cs]: Possible future work: According to GDPR we might want to limit the collection of user account information in auto-collection scenarios and rather resort to anonymous identification. If message contains actual user ID - then it's fine and it's customer's choice to trace it, if we infer actual user ID from the data and store it - we will need to provide ability for customer to know about this data being collected and ability to clear it on backend.

File: src/Adapters/DiagnosticSource/DiagnosticSourceListener.csproj (hunk 1 above)
Comment (SergeyKanzhelev) [reply src/Adapters/DiagnosticSource/DiagnosticSourceListener.csproj]: I'll address it in separate review. Will check the guidelines

File: src/Adapters/Log4NetAppender.Net45/ApplicationInsightsAppender.cs (hunk 3 above)
Comment (SergeyKanzhelev) [reply src/Adapters/Log4NetAppender.Net45/ApplicationInsightsAppender.cs]: Yep. Keeping it as is for now.

File: src/Adapters.Tests/DiagnosticSource.Tests/DiagnosticSourceListener.Tests.csproj (hunk 2 above)
Comment (SergeyKanzhelev) [reply src/Adapters.Tests/DiagnosticSource.Tests/DiagnosticSourceListener.Tests.csproj]: we do not really need `<IncludeAssets>All</IncludeAssets>` as this is default. I'll remove it everywhere

**Pull request merged**